### ISSUE TITLE
End of training Sample Ratio decay

### ISSUE THREAD

**Issue by PeterVennerstrom** [microsoft/SoftTeacher#197]
The paper describes a gradual decay of the sample ratio at the end of training.  

Partially labeled data:
> the data sampling ratio sr is set to 0.2 and gradually decreases to 0 over the last 10k iterations.
Fully labeled data:
>the data sampling ratio sr is set to 0.5 and gradually decreases to 0 in the last 20k iterations.

There are some [commented out lines in semi_sampler.py](https://github.com/microsoft/SoftTeacher/blob/944b1313910a94fa823c7c9f1a0d9ee613f04b49/ssod/datasets/samplers/semi_sampler.py#L196-L199) which look similar.  They suggest updates to the ratio are accomplished by weighting.

How are the 10k and 20k iteration decay points set?

Thanks!

**Comment by zfli-xdu** [comment]
look forward to the answer about sample ratio decay

**Comment by PeterVennerstrom** [comment]
Tested the Weighter hook.

`dict(type="Weighter", steps=[170000, 172500, 175000, 177500], vals=[4, 3, 2, 1, 0], name="unsup_weight")`

It worked as expected.

A potential issue is the learning rate step down doesn't account for less contributions from the unsupervised images in each batch effectively increasing the learning rate on the remaining loss level.  

E.g. at the end there is no loss retained from the unsupervised images so the entire learning rate is applied to the single supervised image.  

Another issue is the number supervised images at the end of training is effectively decreased relative to what is described in the paper unless the number of iterations is increased.  (when using loss weighting rather than adjusting the sample ratio)
